### PR TITLE
Update/38743 loading changes in core profiler

### LINKED ISSUE #38743: Loading changes in core profiler
### Issue description

Reference: Y5pUYSJPsGEud1vknUZhi8-fi-1911_36077

Loading related changes in Core profiler

### Acceptance criteria

1. Instead of a sole loading component shown in the middle, we can try to use button loaders in `Continue` buttons to improve UX. ([screenshot](https://user-images.githubusercontent.com/3747241/246131559-f858e84b-2196-4d61-8dbd-8f55ab69e346.png)) Y5pUYSJPsGEud1vknUZhi8-fi-1440_33442 
2. Extensions screen should be skipped when every plugin is installed
3. The loading sequence after extensions step should include a 3rd loader: Y5pUYSJPsGEud1vknUZhi8-fi-1440_33442
4. Customize loading sequence for skipping setup Y5pUYSJPsGEud1vknUZhi8-fi-1440_33442

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Some minor UI fixes in this PR:
- Spinners no longer appear between pages in the core profiler.
- Spinner was added to the 'Continue' button in the Business Info page, to indicate that it takes a moment to finish its' API calls
- Extensions screen is skipped when all available plugins are installed
- Loading screen after Extensions screen should have a 3rd loader step
- Skipping setup has a new customised loader

Closes #38743  .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate WooCommerce on a brand new site
1. Install the WooCommerce Beta Tester plugin
1. Go to /wp-admin/tools.php?page=woocommerce-admin-test-helper
1. Enable the 'core-profiler' feature flag
1. Visit your public URL /wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard
1. Go through the new core profiler flow.
1. Observe that spinners no longer appear between pages
1. Observe that upon submission of the Business Info page, the button shows a spinner until the page changes
1. Observe that there is a 3rd loader step in the loader during extensions installation
1. After a first runthrough installing all the plugins, a second runthrough should not show the extensions page
1. Clicking on "Skip guided setup" on the first page and going through the skip sequence should show a customised loader now

Extensions installation 3rd loader step:
image

Skip Guided Setup loader:
image

### CODE REVIEW

File: plugins/woocommerce-admin/client/core-profiler/index.tsx [hunk 1]
```diff
@@ -1085,8 +1102,18 @@ export const coreProfilerStateMachineDefinition = createMachine( {
...
+								cond: ( context, event ) => {
+									// Skip the plugins page
+									// When there is 0 plugin returned from the server
+									// Or all the plugins are activated already.
+									return (
+										event.data.length === 0 ||
+										event.data.filter(
+											( plugin: Extension ) =>
+												plugin.is_activated === false
+										).length === 0
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/core-profiler/index.tsx]: Nit: I think we can just replace `filter` with the `every` method, and we don't need to check `length`.

```js
event.data.every( ( plugin ) => plugin.is_activated )
```

File: plugins/woocommerce-admin/client/core-profiler/pages/BusinessInfo.tsx [hunk 2]
```diff
@@ -347,9 +349,14 @@ export const BusinessInfo = ( {
 										geolocationOverruled || false,
 								},
 							} );
+							setHasSubmitted( true );
 						} }
 					>
-						{ __( 'Continue', 'woocommerce' ) }
+						{ hasSubmitted ? (
+							<Spinner />
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/core-profiler/pages/BusinessInfo.tsx:356]: I wonder if an error occurs, will the state be reset?

**Review by chihsuan — APPROVED** [review]
Awesome! Tested well. 👍 

Left some minor comments. Pre-approving!

File: plugins/woocommerce-admin/client/core-profiler/pages/BusinessInfo.tsx (hunk 2 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/core-profiler/pages/BusinessInfo.tsx:356]: It won't, but it'll move to the next state if/when the API call promise rejects

File: plugins/woocommerce-admin/client/core-profiler/index.tsx (hunk 1 above)
Comment (rjchow) [reply plugins/woocommerce-admin/client/core-profiler/index.tsx]: nice, that works!

**Review by adrianduffell — APPROVED** [review]
This tested nice! Thanks @moon0326 and @rjchow 🚀

**Pull request merged**